bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/TableViewer.java

```diff
@@ -506,4 +510,39 @@ void handleExpandableNodeClicked(Widget w) {
 		}
 	}
 
+	@Override
+	void handleKeySelection(KeyEvent e) {
+		if (e.keyCode == SWT.KEYPAD_MULTIPLY || e.keyCode == SWT.KEYPAD_ADD || e.keyCode == SWT.ARROW_RIGHT) {
+			if (e.widget instanceof Table table) {
+				TableItem[] selection = table.getSelection();
+				if (selection.length != 1) {
+					return;
+				}
+				handleExpandableNodeClicked(selection[0], false);
+			}
+		}
+	}
+
+	@Override
+	void handleMenuSelection(MenuEvent e, Menu popupMenu) {
+		if (table.getSelection().length != 1) {
+			return;
+		}
+		TableItem selectedItem = table.getSelection()[0];
+		if (selectedItem.getData() instanceof ExpandableNode) {
+			MenuItem[] items = popupMenu.getItems();
+			for (int i = 0; i < items.length; i++) {
+				items[i].dispose();
+			}
+			MenuItem newItem = new MenuItem(popupMenu, SWT.NONE);
+			newItem.setText("Expand All Remaining Items"); //$NON-NLS-1$
```

Comment:
Two things:
1) The text should be externalized.
2) "Expand All Remaining Items" ->  "Show All Remaining Items"

Reply:
fixed.